packages/frontend-2/components/settings/workspaces/General/DeleteDialog.vue

```diff
@@ -0,0 +1,94 @@
+<template>
+  <LayoutDialog
+    v-model:open="isOpen"
+    title="Delete workspace"
+    max-width="sm"
+    :buttons="dialogButtons"
+  >
+    <p>
+      Are you sure you want to
+      <span class="font-medium">permanently delete</span>
+      the selected workspace?
+    </p>
+    <div
+      class="rounded border bg-foundation-2 border-outline-3 text-body-2xs text-foreground font-medium py-3 px-4 my-4"
+    >
+      {{ workspace.name }}
+    </div>
+    <p>
+      This action
+      <span class="font-medium">cannot</span>
+      be undone.
+    </p>
+  </LayoutDialog>
+</template>
+<script setup lang="ts">
+import { useMutation } from '@vue/apollo-composable'
+import { deleteWorkspaceMutation } from '~/lib/settings/graphql/mutations'
+import type { LayoutDialogButton } from '@speckle/ui-components'
+import { graphql } from '~~/lib/common/generated/gql'
+import {
+  convertThrowIntoFetchResult,
+  getFirstErrorMessage
+} from '~~/lib/common/helpers/graphql'
+import { homeRoute } from '~/lib/common/helpers/route'
+import type { SettingsWorkspaceGeneralDelete_WorkspaceFragment } from '~~/lib/common/generated/gql/graphql'
+import { ToastNotificationType, useGlobalToast } from '~~/lib/common/composables/toast'
+
+graphql(`
+  fragment SettingsWorkspaceGeneralDelete_Workspace on Workspace {
```

Comment:
You aren't asking for this fragment in any other fragments/queries. It works now, cause the other fragments already ask for the same data, but you should always make the dependency explicit by asking for this in the parent component's fragment (SettingsWorkspaceGeneral)

Reply:
It only needs the ID and name, but I think naming the prop `workspace` was a bit misleading, change it be explict `workspaceId` and `workspaceName`